core/chains/evm/client/errors.go

```diff
@@ -167,13 +167,16 @@ var klaytn = ClientErrors{
 // Nethermind
 // All errors: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.TxPool/AcceptTxResult.cs
 // All filters: https://github.com/NethermindEth/nethermind/tree/9b68ec048c65f4b44fb863164c0dec3f7780d820/src/Nethermind/Nethermind.TxPool/Filters
-var nethermindFatal = regexp.MustCompile(`(: |^)(SenderIsContract|Invalid|Int256Overflow|FailedToResolveSender|GasLimitExceeded)$`)
+var nethermindFatal = regexp.MustCompile(`(: |^)(SenderIsContract|Invalid(, transaction Hash is null)?|Int256Overflow|FailedToResolveSender|GasLimitExceeded(, Gas limit: \d+, gas limit of rejected tx: \d+)?|NonceGap(, Future nonce. Expected nonce: \d+)?)$`)
```

Comment:
NonceGap seems very interesting!
I haven't seen such an error being returned before.

I think we need special handling for this. If we treat this as fatal, bad things can happen in our txm.
For example, we send multiple Txs for the same FromAddresses one after another, so lets say we send the following:
Tx1: nonce 1
Tx2: nonce 2

It is possible that a node sees Tx2 before Tx1. In that case, this is not a fatal error, and we should just retry.

For now, I think we should treat NonceGap as a retryable error.

But this is a very critical issue we need to handle here, or else can cause major problems in some nodes.

@samsondav @jmank88 for a second look

Reply:
For context, other eth clients, when they see a nonce gap, just accept the Tx, and keep it in their mempool, and wait till they see the missing nonce too.

Reply:
Ah yeah I guess we normally rely on the queue/pending node and RPC config to be aligned so this is avoided all together... :thinking:

Reply:
Do you mean you could configure an RPC endpoint to not return such NonceGap errors?

In any case, we cannot always control RPCs, so should also handle it in code correctly.

Reply:
I mean that when properly configured, we should never get so far ahead of ourselves that this happens, but yes it would be better to support more cases.
Our docs describe how to align `ETH_MAX_QUEUED_TRANSACTIONS`/`ETH_MAX_IN_FLIGHT_TRANSACTIONS` with the RPC node's `AccountSlots`/`GlobalSlots`/`AccountQueue`/`GlobaQueue`:
https://docs.chain.link/chainlink-nodes/evm-performance-configuration

Reply:
>>when properly configured, we should never get so far ahead of ourselves that this happens,
How did you deduce that nethermind throws this error only when you run too far ahead? Can they not return this error even if you just skip 1-2 nonces?

Reply:
I am not trying to say anything about Nethermind in particular.

Reply:
That's not exactly how it works. Nethermind doesn't error out if you send a transaction with a higher nonce, but rather by sending a tx with a higher nonce than the sum of `current_nonce + pending_txs_in_mempool` ( you can find this behavior here: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.TxPool/Filters/GapNonceFilter.cs#L28 ). 
This is probably to protect against flooding the RPC with txs with high nonces and forcing it to keep them in its state.

To answer @prashantkumar1982 's point, I think the answer comes from how the gap occurred in the first place. I see two cases: same address was used externally or the RPC dropped some of the latest txs in mempool and the next tx in line ended up with a gap nonce. Either way, the nonce order has been messed up, and we should probably error out.
Let me know what you guys think.

Reply:
Is the misconfiguration case still possible too? What error is returned for the first tx after the pool is full?

I am thinking now that some of the edge cases we are worried about here would likely only occur on batch (re)broadcasts, since we send to active primaries serially and should halt as soon as the pool is full... :thinking:

Reply:
Is there another route for _correcting_ the nonce instead of giving up?

Reply:
I **think** it should not be possible for Chainlink to send txes with a nonce gap. I can't say for certain though that there isn't some combination of re-orgs, resending etc where it might not be possible.

Reply:
Example here:

https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/txmgr/eth_resender.go#L148

This orders by eth_tx.id, then nonce. Is it possible you could send nonces out of order? I think so. We would need to review all such cases (perhaps this is the only one?).

Reply:
IMO it's a bug in Nethermind. go-ethereum supports out-of-order transmissions